### DISCUSSION TITLE
Which version of UUID does useId use? (e.g. v1 or v4)

### DISCUSSION THREAD

**Discussion by geoyws** [mantinedev/mantine#591] [Q&A] [answered]
I couldn't find anything in the documentation 😅

**Comment by UltraInstinct05** [comment, marked as answer]
I don't see anywhere in https://mantine.dev/hooks/use-id/ where it mentions that the `uuid` package is being used. Nor is it present in `package.json`.

This is the source for it https://github.com/mantinedev/mantine/blob/master/src/mantine-hooks/src/use-id/use-id.ts. So it either sets the `id` based on the string you pass or else generates a random `mantine-` prefixed string (which uses `Math.random` under the hood).

**Reply by geoyws** [reply]
Apologies I should have just dug into the source code to see. 

https://github.com/mantinedev/mantine/blob/master/src/mantine-hooks/src/utils/random-id/random-id.ts

UUID v1 would be a great way to avoid collisions though.
